test/template_test.rb

```diff
@@ -248,7 +248,18 @@ def test_xss_protection_in_attributes
   end
 
   def test_xss_protection_in_attributes_with_safe_strings
-    assert_equal("<div data-html='<foo>bar</foo>'></div>\n", render('%div{ "data-html" => "<foo>bar</foo>".html_safe }', :action_view))
+    assert_equal("<div data-html='&lt;foo&gt;bar&lt;/foo&gt;'></div>\n", render('%div{ "data-html" => "<foo>bar</foo>".html_safe }', :action_view))
+    assert_equal(<<-HTML, render(<<-HAML, :action_view))
+<meta content='&#39;&quot;' />
+<meta content='&#39;&quot;' />
+<meta content='&#39;&quot;' />
+HTML
+%meta{ content: %{'"}.html_safe }
+- val = %{'"}.html_safe
+%meta{ content: val }
+- hash = { content: val }
+%meta{ hash }
+HAML
```

Comment:
Can you pull this out into actual template files, instead of having this pasted into this already-large file? Should be as easy as creating some files in the `templates` and `results` folder.

I'd really like to get rid of these assertion-based tests in this test suite that is focused on full-template file comparison.

Reply:
done https://github.com/haml/haml/pull/1028/commits/da0e7da5c89a54f067427ed7ece5e15eb3e5a178